clients/lsp-lua.el

```diff
@@ -556,6 +556,87 @@ and `../lib` ,exclude `../lib/temp`.
+  "Location of Lua Language Server main.lua (latest)."
+  :group 'lsp-lua-language-server-latest
+  :risky t
+  :type 'file)
+
+(defcustom lsp-clients-lua-language-server-latest-args '("-E")
+  "Arguments to run the Lua Language server (latest)."
+  :group 'lsp-lua-language-server-latest
+  :risky t
+  :type '(repeat string))
+
+(defun lsp--find-latest-lua-ls-release-url ()
+  "Fetch the latest version of lua-language-server from its github repo
+(for use with `lsp-lua-language-server-install'). Search for \"linux-x64\",
+\"darwin-x64\" and \"win32-x64\" tags on releases."
+  (let ((system-suffix (pcase system-type
+                         ('gnu/linux "linux-x64")
+                         ('darwin "darwin-x64")
+                         ('windows-nt "win32-x64")
+                         (_ "linux-x64")))
+        (url-request-method "GET"))
+    (with-current-buffer (url-retrieve-synchronously "https://api.github.com/repos/sumneko/lua-language-server/releases/latest")
+      ;; Go to beginning of the last line (where the retrieved json with information
+      ;; is in the buffer from `url-retrieve-synchronously')
+      (goto-char (point-max))
+      (goto-char (point-at-bol))
+      (let* ((json-result (prog1 (json-read) (kill-buffer)))
+             (os-search-pred (lambda (json-entry)
+                      (string-match-p system-suffix (alist-get 'name json-entry)))))
+        (message "Latest version found: %s" (alist-get 'name json-result))
+        (alist-get 'browser_download_url (seq-find os-search-pred (alist-get 'assets json-result)))
+        ))))
+
+(defun lsp-lua-language-server-install-latest (client callback error-callback update?)
+  "Download the latest version of lua-language-server and extract it to
+`lsp-lua-language-server-install-dir'."
+  (ignore client update?)
+  (let ((store-path (expand-file-name "lua-language-server-github" lsp-clients-lua-language-server-install-dir)))
+    (lsp-download-install
+     (lambda (&rest _)
+       (set-file-modes lsp-clients-lua-language-server-latest-bin #o0700)
+       (funcall callback))
+     error-callback
+     :url (lsp--find-latest-lua-ls-release-url)
+     :store-path store-path
+     :decompress (pcase system-type ('windows-nt :zip) (_ :targz)))))
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-stdio-connection (lambda () (or lsp-clients-lua-language-server-command
+                                                       `(,lsp-clients-lua-language-server-latest-bin
+                                                         ,@lsp-clients-lua-language-server-latest-args
+                                                         ,lsp-clients-lua-language-server-latest-main-location)))
+                                        '(lambda () (and (f-exists? lsp-clients-lua-language-server-latest-bin) (f-exists? lsp-clients-lua-language-server-latest-main-location))))
+  :major-modes '(lua-mode)
+  :priority -2
+  :server-id 'lua-language-server-latest
```

Comment:
can you modify the existing client instead of creating a new one? You can try first the existing variables and if they are not present.